cmd/wallet/deposits.go

```diff
@@ -0,0 +1,189 @@
+	if len(walletStr) > 0 {
+		walletPublicKeyHash, err := hexToWalletPublicKeyHash(walletStr)
+		if err != nil {
+			return []depositEntry{}, fmt.Errorf("failed to extract wallet public key hash: %v", err)
+		}
+
+		filter.WalletPublicKeyHash = [][20]byte{walletPublicKeyHash}
+	}
+
+	depositRevealedEvent, err := tbtcChain.PastDepositRevealedEvents(filter)
+	if err != nil {
+		return []depositEntry{}, fmt.Errorf(
+			"failed to get past deposit revealed events: [%w]",
+			err,
+		)
+	}
+
+	logger.Infof("found %d DepositRevealed events", len(depositRevealedEvent))
+
+	result := make([]depositEntry, len(depositRevealedEvent))
+	for i, event := range depositRevealedEvent {
+		logger.Debugf("getting details of deposit %d/%d", i+1, len(depositRevealedEvent))
+
+		depositKey := buildDepositKey(event.FundingTxHash, event.FundingOutputIndex)
+
+		depositRequest, err := tbtcChain.GetDepositRequest(event.FundingTxHash, event.FundingOutputIndex)
+		if err != nil {
+			return result, fmt.Errorf(
+				"failed to get deposit request: [%w]",
+				err,
+			)
+		}
+
+		result[i] = depositEntry{
+			wallet:                        event.WalletPublicKeyHash,
+			depositKey:                    depositKey,
+			revealedAtBlock:               event.BlockNumber,
+			isSwept:                       depositRequest.SweptAt.After(depositRequest.RevealedAt),
+			fundingTransactionHash:        event.FundingTxHash,
+			fundingTransactionOutputIndex: event.FundingOutputIndex,
+			amountBtc:                     convertSatToBtc(float64(depositRequest.Amount)),
+		}
+	}
+
+	return result, nil
+}
+
+func printTable(deposits []depositEntry) error {
+	w := tabwriter.NewWriter(os.Stdout, 2, 4, 1, ' ', tabwriter.AlignRight)
+	fmt.Fprintf(w, "index\twallet\tvalue (BTC)\tdeposit key\tfunding transaction\tswept\t\n")
+
+	for i, deposit := range deposits {
+		fmt.Fprintf(w, "%d\t%s\t%.5f\t%s\t%s\t%v\t\n",
+			i,
+			"0x"+hex.EncodeToString(deposit.wallet[:]),
+			deposit.amountBtc,
+			deposit.depositKey,
+			fmt.Sprintf(
+				"%s:%d",
+				deposit.fundingTransactionHash.Hex(bitcoin.ReversedByteOrder),
+				deposit.fundingTransactionOutputIndex,
+			),
+			deposit.isSwept,
+		)
+	}
+
+	if err := w.Flush(); err != nil {
+		return fmt.Errorf("failed to flush the writer: %v", err)
+	}
+
+	return nil
+}
+
+func hexToWalletPublicKeyHash(str string) ([20]byte, error) {
+	walletHex, err := hexutil.Decode(str)
+	if err != nil {
+		return [20]byte{}, fmt.Errorf("failed to parse arguments: %w", err)
+	}
+
+	var walletPublicKeyHash [20]byte
+	copy(walletPublicKeyHash[:], walletHex)
+
+	return walletPublicKeyHash, nil
+}
+
+func buildDepositKey(
+	fundingTxHash bitcoin.Hash,
+	fundingOutputIndex uint32,
+) string {
+	fundingOutputIndexBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(fundingOutputIndexBytes, fundingOutputIndex)
+
+	depositKey := crypto.Keccak256Hash(
+		append(fundingTxHash[:], fundingOutputIndexBytes...),
+	)
+
+	return depositKey.Hex()
+}
```

Comment:
There is a `buildDepositKey` in `pkg/chain/ethereum/tbtc.go` and we are passing a handle to `tbtc.Chain` to `ListDeposits`. The code for building a deposit key is chain-specific so I would consider exposing a function in `BridgeChain interface` of `pkg/tbtc/chain.go` to `BuildDepositKey`.

Reply:
Agreed. This will allow us to get rid of the `github.com/ethereum/go-ethereum` import altogether.

Reply:
I did it in an initial version but then decided to just copy the `buildDepositKey` function as I thought that we will soon get rid of the CLI in favor of automated proposals and won't need the exported `BuildDepositKey` in the interface.

Reply:
IMO, we should do it anyway. Even if we automate it, this code will live in a chain-agnostic package (either `pkg/tbtc` or a dedicated one like `pkg/coordinator`). That means we would still need this function to be exposed by the chain interface.

Reply:
6e4e90ad3ba3b0bd864ca0ccf991ee7392295450